### PR TITLE
[Text Analytics] Implement `IsEnvironmentReadyAsync`

### LINKED ISSUE #35452: [Text Analytics] Fix tests failing with "401 (PermissionDenied)"
This exclusively affects tests running against the dynamic resource. The root cause is probably that tests start running before the resource has been fully created and stabilized.

```text
Azure.RequestFailedException : Access denied due to invalid subscription key or wrong API endpoint. Make sure to provide a valid key for an active subscription and use a correct regional API endpoint for your resource.
Status: 401 (PermissionDenied)
ErrorCode: 401
```

### PR DESCRIPTION
**Pull request by joseharriaga**
Resolves: https://github.com/Azure/azure-sdk-for-net/issues/35452.

Implemented `IsEnvironmentReadyAsync` with a simple request to the dynamic resource to confirm that permissions have propagated successfully, because otherwise we're at risk of having a bunch of test failures like this one:

```text
Azure.RequestFailedException : Access denied due to invalid subscription key or wrong API endpoint. Make sure to provide a valid key for an active subscription and use a correct regional API endpoint for your resource.
Status: 401 (PermissionDenied)
ErrorCode: 401
```

Additionally, while testing, I found another instance of the transient Internal Server Error. The difference is that the message is simply "Internal Server Error." while the previous known case said "Failed to process task after several retry":

```text
Azure.RequestFailedException : Internal Server Error.
Status: 200 (OK)
ErrorCode: InternalServerError

Additional Information:
Target: #/tasks/items/1
error-0: InternalServerError: Internal Server Error.
```

I added this one to the `ShouldRetry` method of `RetryOnInternalServerErrorAttribute`.

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RetryOnInternalServerErrorAttributeTests.cs [hunk 1]
```diff
@@ -14,18 +14,24 @@ namespace Azure.AI.TextAnalytics.Tests
     public class RetryOnInternalServerErrorAttributeTests
     {
         private const string FailedToProcessTaskExceptionMessage =
-            "Failed to process task after several retry,"
-            + " Status: 200 (OK),"
-            + " ErrorCode: InternalServerError";
+            "Failed to process task after several retry"
```
Comment (heaths) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/RetryOnInternalServerErrorAttributeTests.cs:17]: "With great power comes great responsibility." :)

Have you been letting the service team know about these issues? When I initially did something like this with Key Vault, it was after effectively triaging the sporadic issues we were finding and investigations on their side.

**Review by heaths — APPROVED** [review]

**Review by deyaaeldeen — APPROVED** [review]
Looks good, however, the issue seems to be not TA specific. Should the pipeline add some latency after the resource creation step to mitigate it?

**Comment by heaths** [comment]
> Looks good, however, the issue seems to be not TA specific. Should the pipeline add some latency after the resource creation step to mitigate it?

We don't want to add latency to all pipelines. Few, hopefully, need it. See my review for what I suggested to inject latency when needed.

**Review by mssfang — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RetryOnInternalServerErrorAttributeTests.cs (hunk 1 above)
Comment (joseharriaga) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/RetryOnInternalServerErrorAttributeTests.cs:17]: Yes, I filed bugs for each of these and provided all the information that I have.

**Pull request merged**